GameServerLib/GameObjects/AttackableUnits/AI/Minion.cs

```diff
@@ -101,32 +101,36 @@ protected bool ScanForTargets()
 
             IAttackableUnit nextTarget = null;
             var nextTargetPriority = 14;
-            var objects = _game.ObjectManager.GetObjects();
+            // TODO: Verify if we still need to order this by distance (if it is ordered by time created).
+            var nearestObjects = _game.Map.CollisionHandler.QuadDynamic.GetNearestObjects(this);
```

Comment:
I think you should order it by distance as I don't find a reason the objects will be ordered by distance. Also, I think it can omit some of the farthest objects from time to time (depends on the subdivisions).